source/common/http/conn_manager_impl.cc

```diff
@@ -396,6 +396,25 @@ ConnectionManagerImpl::ActiveStream::~ActiveStream() {
   ASSERT(state_.filter_call_state_ == 0);
 }
 
+void ConnectionManagerImpl::ActiveStream::resetIdleTimer() {
+  if (idle_timer_ != nullptr) {
+    idle_timer_->enableTimer(idle_timeout_ms_);
```

Comment:
Have you tried a loadtest run with this on?  I believe internally we ended up needing to latch "time of last data" and when the alarm fired resetting the alarm if necessary because the adds/removes were expensive enough we couldn't afford to thrash.  I don't know if Envoy has a more efficient alarm management under the hood so I'm unsure if we'll have the same problem here.

Realistically if we leave it opt in or high timeouts, you're going to be the one who tracks down any perf problem so I'm fine with you landing first and seeing if it's a problem :-P

Reply:
I discussed this with @mattklein123 earlier; we agreed that libevent timers are probably fine for now, and we should leave this as a TODO for later optimization.

Reply:
I've thought about this (because I added idle-timeout to tcp proxy, with the same perf concern). I think we could probably make this fast enough if we add a tunable to timers for inaccuracy-percent or something. Then keep track of the current time it will expire, look at the new time, and if it has moved by less than X%, don't reschedule the timer. 

But I decided not to bother until rescheduling timers showed up hot in a profile that I cared about.

Reply:
It's possible/likely that on super high throughput streaming case this timer reset behavior is going to show up on perf traces, but TBH I don't know how much and yeah I think it's probably fine for now. I would probably go for the simple approach first and we can make better later if needed?

Reply:
SGTM.  As said Harvey ends up doing most of our perf tuning so if it shows up I'm sure he'll just deal with it later :-)